sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1494,6 +1493,56 @@ class Dataset[T] private[sql](
   }
 
   /**
+   * Returns a new [[Dataset]] by sampling a fixed num of rows, using a random seed.
+   *
+   * @param withReplacement Sample with replacement or not.
+   * @param num num of rows to generate.
+   * @param seed Seed for sampling
+   * @return the sampling [[Dataset]]
+   */
+  def takeSample(withReplacement: Boolean, num: Int, seed: Long): Dataset[T] = {
+    val numStDev = 10.0
+    require(num >= 0, "Negative number of elements requested")
+    require(num <= (Int.MaxValue - (numStDev * math.sqrt(Int.MaxValue)).toInt),
+      "Cannot support a sample size > Int.MaxValue - " +
+        s"$numStDev * math.sqrt(Int.MaxValue)")
+    if (num == 0) {
+      sqlContext.createDataset[T](Seq())
+    } else {
+      val initialCount = this.count()
```

Comment:
This can be a non-trivial call. It will force us to evaluate a significant part of the query.